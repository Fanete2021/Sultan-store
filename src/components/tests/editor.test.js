/* eslint-disable testing-library/render-result-naming-convention */
const React = require('react')
const Editor = require('../Editor').default
const { selectCategories, selectEditor } = require('../../store/reducers/Selectors')
const { productState } = require('./states')
const { render, screen, fireEvent } = require('@testing-library/react')
const { cartSlice } = require('../../store/reducers/slices/CartSlice')
const { editorSlice } = require('../../store/reducers/slices/EditorSlice')
const { productSlice } = require('../../store/reducers/slices/ProductSlice')
const reduxHooks = require('react-redux')
const { BrowserRouter } = require('react-router-dom')

jest.mock('../../store/reducers/Selectors')
jest.mock('react-redux')

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')

describe('editor', () => {
    it('should create Editor', () => {
        selectCategories.mockReturnValue({ categories: [] })
        selectEditor.mockReturnValue({ product: null })

        const editor = render(<Editor />)

        expect(editor).toMatchSnapshot()
    })

    it('should create Editor with products', () => {
        selectCategories.mockReturnValue({ categories: [] })
        selectEditor.mockReturnValue({ product: productState })

        const editor = render(<Editor />)

        expect(editor).toMatchSnapshot()
    })

    it('should dispatch actions', () => {
        const dispatch = jest.fn()
        mockedDispatch.mockReturnValue(dispatch)

        selectCategories.mockReturnValue({ categories: [] })
        selectEditor.mockReturnValue({ product: productState })

        const mockedHideEditorComplete = jest.spyOn(editorSlice.actions, 'changeShowing')

        const mockedChangeProductComplete = jest.spyOn(productSlice.actions, 'changeProduct')

        const mockedDeleteProductComplete = jest.spyOn(productSlice.actions, 'deleteProductById')
        const mockedDeleteItemFromCartComplete = jest.spyOn(cartSlice.actions, 'deleteItemById')
        const mockedDeleteProductFromEditorComplete = jest.spyOn(
            editorSlice.actions,
            'deleteEditableProduct'
        )

        render(
            <BrowserRouter>
                <Editor />
            </BrowserRouter>
        )

        fireEvent.click(screen.getByTestId('save'))
        fireEvent.click(screen.getByTestId('trash'))

        expect(dispatch).toHaveBeenCalledTimes(6)

        expect(mockedHideEditorComplete).toHaveBeenCalledWith(false)
        expect(mockedChangeProductComplete).toHaveBeenCalledWith(productState)

        expect(mockedHideEditorComplete).toHaveBeenCalledWith(false)
        expect(mockedDeleteProductComplete).toHaveBeenCalledWith(productState.id)
        expect(mockedDeleteItemFromCartComplete).toHaveBeenCalledWith(productState.id)
        expect(mockedDeleteProductFromEditorComplete).toHaveBeenCalledWith()
    })
})
