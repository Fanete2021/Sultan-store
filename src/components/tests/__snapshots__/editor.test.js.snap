// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor should create Editor 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="editor__content"
        >
          <img
            alt=""
            class="editor__close"
            src=""
          />
          <div
            class="editor__inputs"
          >
            <div
              class="inputs__url-image"
            >
              <span>
                Url: 
              </span>
              <input
                value=""
              />
              <div
                class="url-image__content"
              >
                <img
                  alt=""
                  src=""
                />
              </div>
            </div>
            <div
              class="inputs__size"
            >
              <span>
                Размер: 
              </span>
              <input
                value=""
              />
              <select>
                <option
                  value="г"
                >
                  г
                </option>
                <option
                  selected=""
                  value="мл"
                >
                  мл
                </option>
              </select>
            </div>
            <div
              class="inputs__title"
            >
              <span>
                Название: 
              </span>
              <input
                value=""
              />
            </div>
            <div
              class="inputs__description"
            >
              <span>
                Описание: 
              </span>
              <input
                value=""
              />
            </div>
            <div
              class="inputs__barcode"
            >
              <span>
                Штрихкод: 
              </span>
              <input
                value=""
              />
            </div>
            <div
              class="inputs__manufacturer"
            >
              <span>
                Производитель: 
              </span>
              <input
                value=""
              />
            </div>
            <div
              class="inputs__brand"
            >
              <span>
                Бренд: 
              </span>
              <input
                value=""
              />
            </div>
            <div
              class="inputs__price"
            >
              <span>
                Цена: 
              </span>
              <input
                value="0"
              />
            </div>
            <div
              class="inputs__categories"
            >
              <div
                class="categories__title"
              >
                Категории: 
              </div>
            </div>
          </div>
          <div
            class="editor__actions"
          >
            <div
              class="actions__save"
              data-testid="save"
            >
              <img
                alt=""
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="editor__content"
      >
        <img
          alt=""
          class="editor__close"
          src=""
        />
        <div
          class="editor__inputs"
        >
          <div
            class="inputs__url-image"
          >
            <span>
              Url: 
            </span>
            <input
              value=""
            />
            <div
              class="url-image__content"
            >
              <img
                alt=""
                src=""
              />
            </div>
          </div>
          <div
            class="inputs__size"
          >
            <span>
              Размер: 
            </span>
            <input
              value=""
            />
            <select>
              <option
                value="г"
              >
                г
              </option>
              <option
                selected=""
                value="мл"
              >
                мл
              </option>
            </select>
          </div>
          <div
            class="inputs__title"
          >
            <span>
              Название: 
            </span>
            <input
              value=""
            />
          </div>
          <div
            class="inputs__description"
          >
            <span>
              Описание: 
            </span>
            <input
              value=""
            />
          </div>
          <div
            class="inputs__barcode"
          >
            <span>
              Штрихкод: 
            </span>
            <input
              value=""
            />
          </div>
          <div
            class="inputs__manufacturer"
          >
            <span>
              Производитель: 
            </span>
            <input
              value=""
            />
          </div>
          <div
            class="inputs__brand"
          >
            <span>
              Бренд: 
            </span>
            <input
              value=""
            />
          </div>
          <div
            class="inputs__price"
          >
            <span>
              Цена: 
            </span>
            <input
              value="0"
            />
          </div>
          <div
            class="inputs__categories"
          >
            <div
              class="categories__title"
            >
              Категории: 
            </div>
          </div>
        </div>
        <div
          class="editor__actions"
        >
          <div
            class="actions__save"
            data-testid="save"
          >
            <img
              alt=""
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`editor should create Editor with products 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="editor__content"
        >
          <img
            alt=""
            class="editor__close"
            src=""
          />
          <div
            class="editor__inputs"
          >
            <div
              class="inputs__url-image"
            >
              <span>
                Url: 
              </span>
              <input
                value=""
              />
              <div
                class="url-image__content"
              >
                <img
                  alt=""
                  src=""
                />
              </div>
            </div>
            <div
              class="inputs__size"
            >
              <span>
                Размер: 
              </span>
              <input
                value="22"
              />
              <select>
                <option
                  selected=""
                  value="г"
                >
                  г
                </option>
                <option
                  value="мл"
                >
                  мл
                </option>
              </select>
            </div>
            <div
              class="inputs__title"
            >
              <span>
                Название: 
              </span>
              <input
                value="AOS"
              />
            </div>
            <div
              class="inputs__description"
            >
              <span>
                Описание: 
              </span>
              <input
                value="AOS"
              />
            </div>
            <div
              class="inputs__barcode"
            >
              <span>
                Штрихкод: 
              </span>
              <input
                value="1231314"
              />
            </div>
            <div
              class="inputs__manufacturer"
            >
              <span>
                Производитель: 
              </span>
              <input
                value="AOS"
              />
            </div>
            <div
              class="inputs__brand"
            >
              <span>
                Бренд: 
              </span>
              <input
                value="AOS"
              />
            </div>
            <div
              class="inputs__price"
            >
              <span>
                Цена: 
              </span>
              <input
                value="100"
              />
            </div>
            <div
              class="inputs__categories"
            >
              <div
                class="categories__title"
              >
                Категории: 
              </div>
            </div>
          </div>
          <div
            class="editor__actions"
          >
            <div
              class="actions__save"
              data-testid="save"
            >
              <img
                alt=""
                src=""
              />
            </div>
            <div
              class="actions__delete"
              data-testid="trash"
            >
              <img
                alt=""
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="editor__content"
      >
        <img
          alt=""
          class="editor__close"
          src=""
        />
        <div
          class="editor__inputs"
        >
          <div
            class="inputs__url-image"
          >
            <span>
              Url: 
            </span>
            <input
              value=""
            />
            <div
              class="url-image__content"
            >
              <img
                alt=""
                src=""
              />
            </div>
          </div>
          <div
            class="inputs__size"
          >
            <span>
              Размер: 
            </span>
            <input
              value="22"
            />
            <select>
              <option
                selected=""
                value="г"
              >
                г
              </option>
              <option
                value="мл"
              >
                мл
              </option>
            </select>
          </div>
          <div
            class="inputs__title"
          >
            <span>
              Название: 
            </span>
            <input
              value="AOS"
            />
          </div>
          <div
            class="inputs__description"
          >
            <span>
              Описание: 
            </span>
            <input
              value="AOS"
            />
          </div>
          <div
            class="inputs__barcode"
          >
            <span>
              Штрихкод: 
            </span>
            <input
              value="1231314"
            />
          </div>
          <div
            class="inputs__manufacturer"
          >
            <span>
              Производитель: 
            </span>
            <input
              value="AOS"
            />
          </div>
          <div
            class="inputs__brand"
          >
            <span>
              Бренд: 
            </span>
            <input
              value="AOS"
            />
          </div>
          <div
            class="inputs__price"
          >
            <span>
              Цена: 
            </span>
            <input
              value="100"
            />
          </div>
          <div
            class="inputs__categories"
          >
            <div
              class="categories__title"
            >
              Категории: 
            </div>
          </div>
        </div>
        <div
          class="editor__actions"
        >
          <div
            class="actions__save"
            data-testid="save"
          >
            <img
              alt=""
              src=""
            />
          </div>
          <div
            class="actions__delete"
            data-testid="trash"
          >
            <img
              alt=""
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
